src/scripts/clientUtils.js

```diff
@@ -1,35 +1,42 @@
+import * as webSettings from 'webSettings';
 
 export function getCurrentUser() {
     return window.ApiClient.getCurrentUser(false);
 }
 
-//TODO: investigate url prefix support for serverAddress function
-export function serverAddress() {
-    if (AppInfo.isNativeApp) {
-        const apiClient = window.ApiClient;
-
-        if (apiClient) {
-            return apiClient.serverAddress();
-        }
+// TODO: investigate url prefix support for serverAddress function
+export async function serverAddress() {
+    const apiClient = window.ApiClient;
 
-        return null;
+    if (apiClient) {
+        return Promise.resolve(apiClient.serverAddress());
     }
 
-    const urlLower = window.location.href.toLowerCase();
-    const index = urlLower.lastIndexOf('/web');
+    const current = await window.connectionManager.getAvailableServers().then(servers => {
+        if (servers.length !== 0) {
+            return Promise.resolve(servers[0].ManualAddress);
+        }
+    });
 
-    if (index != -1) {
-        return urlLower.substring(0, index);
-    }
+    if (current) return Promise.resolve(current);
 
-    const loc = window.location;
-    let address = loc.protocol + '//' + loc.hostname;
+    const urls = [];
+    urls.push(window.location.origin);
+    urls.push(`${window.location.protocol}//${window.location.hostname}:8096`);
+    urls.push(await webSettings.getServers());
```

Comment:
_Untested_
```suggestion
    urls.push.apply(urls, await webSettings.getServers());
```
Otherwise, server array will be pushed as an element.